src/rbs_location.c

```diff
@@ -15,30 +15,14 @@ void rbs_loc_alloc_children(rbs_allocator_t *allocator, rbs_location_t *loc, int
   loc->children->cap = capacity;
 }
 
-void rbs_loc_add_optional_child(rbs_allocator_t *allocator, rbs_location_t *loc, rbs_constant_id_t name, range r) {
-  // If the location has no children, allocate memory for one child.
-  if (loc->children == NULL) {
-    rbs_loc_alloc_children(allocator, loc, 1);
-  } else if (loc->children->len == loc->children->cap) {
-    // If the location has no more capacity, double the capacity.
-    size_t old_size = RBS_LOC_CHILDREN_SIZE(loc->children->cap);
-    size_t new_size = old_size * 2;
-
-    // Reallocate the children array to the new size.
-    rbs_loc_children *new_children = rbs_allocator_realloc(allocator, loc->children, old_size, new_size, rbs_loc_children);
-    new_children->cap = new_size;
-    loc->children = new_children;
-  }
-
-  // Add a new child to the location.
+void rbs_loc_add_optional_child(rbs_location_t *loc, rbs_constant_id_t name, range r) {
   unsigned short i = loc->children->len++;
```

Comment:
Can you add an assertion here:

```suggestion
  assert(loc->children != NULL && "All children should have been pre-allocated with rbs_loc_alloc_children()");
  assert((loc->children->len + 1 <= loc->children->cap) && "Not enough space was pre-allocated for the children.");
  
  unsigned short i = loc->children->len++;
```